### PR TITLE
[15.0][MIG] sale_order_global_stock_route: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc@Tecnativa TT36984

@chienandalu @CarlosRoca13 please review!

### CODE REVIEW

File: sale_order_global_stock_route/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if "route_id" in vals:
+            lines = self.mapped("order_line").filtered(
+                lambda l: l.route_id.id != vals["route_id"]
+            )
+            lines.write({"route_id": vals["route_id"]})
+        return res
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
```
Comment (rousseldenis) [inline sale_order_global_stock_route/models/sale_order.py]: please put one model per file

File: sale_order_global_stock_route/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        help="When you change this field all the lines will be changed."
+        " After use it you will be able to change each line.",
+    )
+
+    @api.onchange("route_id")
+    def _onchange_route_id(self):
+        """We could do sale order line route_id field compute store writable.
+        But this field is created by Odoo so I prefer not modify it.
+        """
+        for line in self.order_line:
```
Comment (rousseldenis) [inline sale_order_global_stock_route/models/sale_order.py]: As onchange is always applied on a single record (see https://github.com/odoo/odoo/blob/15.0/odoo/models.py#L6489)

, I would do:

```suggestion
        self.order_line.update({'route_id': self.order_id.route_id})
```

File: sale_order_global_stock_route/tests/test_sale_order_global_stock_route.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class SaleOrderGlobalStockRouteTest(TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline sale_order_global_stock_route/tests/test_sale_order_global_stock_route.py]: Please change this to classmethod

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: sale_order_global_stock_route/models/sale_order.py (hunk 1 above)
Comment (stefan-tecnativa) [reply sale_order_global_stock_route/models/sale_order.py]: Done

File: sale_order_global_stock_route/models/sale_order.py (hunk 2 above)
Comment (stefan-tecnativa) [reply sale_order_global_stock_route/models/sale_order.py]: I didn't know it worked like this, thanks for the lesson learned.

File: sale_order_global_stock_route/tests/test_sale_order_global_stock_route.py (hunk 3 above)
Comment (stefan-tecnativa) [reply sale_order_global_stock_route/tests/test_sale_order_global_stock_route.py]: Done, thanks for all the suggestions

File: sale_order_global_stock_route/models/sale_order.py (hunk 1 above)
Comment (chienandalu) [reply sale_order_global_stock_route/models/sale_order.py]: With one2many relations our guidelines are to keep the model along with the parent model stuff. Please reverse this change

Comment (stefan-tecnativa) [reply sale_order_global_stock_route/models/sale_order.py]: Understood

**Comment by pablontura** [comment]
Functional review. If I try to replace a product or access the RMA smartbutton in the original shipment, I get these errors:

reemplazo

Smartbutton

**Comment by chienandalu** [comment]
@pablontura  You can clearly see in your traceback that the issue is in the `rma` code

**Comment by pablontura** [comment]
> @pablontura You can clearly in your traceback see that the issue is in the `rma` code

Omg sorry! obviously I have commented in the wrong PR... I had both open and I must have been confused

File: sale_order_global_stock_route/models/sale_order.py (hunk 2 above)
Comment (rousseldenis) [reply sale_order_global_stock_route/models/sale_order.py]: @stefan-tecnativa No problem. The most performances gain can be earned when you have several attributes to modify as you can do `recordset.update({<attribute 1>: value, <attribute 2>: value, ...})`

**Comment by chienandalu** [comment]
> Omg sorry! obviously I have commented in the wrong PR... I had both open and I must have been confused

Nevermind :slightly_smiling_face:

File: sale_order_global_stock_route/models/sale_order.py (hunk 2 above)
Comment (chienandalu) [reply sale_order_global_stock_route/models/sale_order.py]: For the present case, I'd go for a direct attribution. It makes no difference at ORM level and the code is much more expresive:
```python
        self.order_line.route_id = self.route_id
```

**Review by rousseldenis — APPROVED** [review]

File: sale_order_global_stock_route/models/sale_order.py (hunk 2 above)
Comment (stefan-tecnativa) [reply sale_order_global_stock_route/models/sale_order.py]: I am agree with you.

**Review by chienandalu — APPROVED** [review]

**Pull request merged**